### PR TITLE
[webhook] Add env from vault path

### PR DESCRIPTION
**Pull request by rocpatel**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | mentioned in #1098 
| License         | Apache 2.0

### What's in this PR?
This PR adds an annotation `vault-env-from-path`, which will mutate a pod with an additional environment variable `VAULT_ENV_FROM_PATH`.  This annotation will hold a comma-delimited list of the vault paths to read from.  When this environment variable is present the `vault-env` binary will call the `InjectSecretsFromVaultPath` method on the `injector` package.  The `InjectSecretsFromVaultPath` method will inject all the secrets from the vault paths into the environment.  It does not sanitize the keys but adds them as is to the environment.  Also, if a later path has a duplicate key to a previous path, that key will be overwritten by the later path value.  This PR probably can be refined a bit more, but wanted to put something together to start a discussion.

### Additional context
I have added some unit tests to test the functionality.  The only thing I don't have is an error if the path doesn't exist.  I also do not take into account the transit encryption for the paths.

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [ ] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### To Do
- [x] If the PR is not complete but you want to discuss the approach, list what remains to be done here

* mutation of configmaps/secrets
* refactoring of code to make it DRY
* possibly uppercasing keys when adding to environment variables
* updating of documentation, need some direction as to where that is
* error handling around missing paths to make it easier to debug missing path issues

### CODE REVIEW

**Comment by bonifaido** [comment]
Thanks, I will check this one today! 👍

File: internal/injector/inejctor.go [hunk 1]
```diff
@@ -215,3 +215,81 @@ func (i SecretInjector) InjectSecretsFromVault(references map[string]string, inj
 
 	return nil
 }
+
+func (i SecretInjector) InjectSecretsFromVaultPath(paths string, inject SecretInjectorFunc) error {
```
Comment (bonifaido) [inline internal/injector/inejctor.go:219]: Basically this function is the same as the original `InjectSecretsFromVault`, just wrapped inside another `for` loop, am I right? We should reuse the same logic extracted from there I think.

Comment (bonifaido) [reply internal/injector/inejctor.go:219]: Really nice tests though! 👍

Comment (rocpatel) [reply internal/injector/inejctor.go:219]: Yes, as I was putting it together, noticed that it's doing the same thing. I'll look into refactoring it a bit

Comment (rocpatel) [reply internal/injector/inejctor.go:172]: Ok, so i went ahead and refactored it a bit.  I didn't find it as simple as just re-using InjectSecretsFromVault with extra parameters.  However, I did pull out some common logic to a new method.  Let me know what you think

**Review by bonifaido — DISMISSED** [review]
LGTM, my only question is regarding this comment:
> The only thing I don't have is an error if the path doesn't exist.

Do we still miss this?

**Review by pbalogh-sa — APPROVED** [review]
Thanks @rocpatel !

**Comment by rocpatel** [comment]
> LGTM, my only question is regarding this comment:
> 
> > The only thing I don't have is an error if the path doesn't exist.
> 
> Do we still miss this?

It was actually doing a hard fail when the path is missing.  Just pushed up something that uses the same logic to check `IgnoreMissingSecrets`.  Also, is there a spot to update the documentation?  Would need to add the extra annotation.

**Review by bonifaido — APPROVED** [review]
LGTM, thank you very much @rocpatel 

The documentation is available at: https://github.com/banzaicloud/bank-vaults-docs

**Pull request merged**